expertise/service/__init__.py

```diff
@@ -62,20 +65,94 @@ def create_app(config=None):
     from . import routes
     app.register_blueprint(routes.BLUEPRINT)
 
-    return app
+    # Any jobs that are running must be marked with cancel
+    redis_config_pool, _ = create_redis(app)
+    redis = RedisDatabase(
+        connection_pool=redis_config_pool
+    )
+    app.logger.info('Running server startup code')
+    for config in redis.load_all_jobs('~Super_User1'):
+        status = config.status
+        if status in [
+            JobStatus.QUEUED,
+            JobStatus.FETCHING_DATA,
+            JobStatus.EXPERTISE_QUEUED,
+            JobStatus.RUN_EXPERTISE
+        ]:
+            app.logger.info(f"{config.job_id} was running - canceling job")
+            config.status = JobStatus.ERROR
+            config.description = 'Server restarted while job was running'
+            config.mdate = int(time.time() * 1000)
+            redis.save_job(config)
 
+    return app
 
 def create_celery(app):
     """
     Initializes a celery application using Flask App
     """
+
+    def set_config_error(job_config):
+        status = job_config.status
+        if status in [
+            JobStatus.QUEUED,
+            JobStatus.FETCHING_DATA,
+            JobStatus.EXPERTISE_QUEUED,
+            JobStatus.RUN_EXPERTISE
+        ]:
+            app.logger.info(f"{job_config.job_id} was running - canceling job")
+            job_config.status = JobStatus.ERROR
```

Comment:
Shouldn't the status be "Cancelled"? I would like to distinguish it from error.

Reply:
Does the "canceled" status make the process functions fail and clears the celery queue?

Reply:
I was thinking about this - I did create a Cancel status but openreview-py will only fail if it sees an "Error", otherwise the process function will wait a day before it times out. I can make it canceled but we would need to update openreview-py to catch the "Cancel" status.